src/services/services.ts

```diff
@@ -428,7 +438,7 @@ class NodeObject implements Node {
 
     public getChildren(sourceFile?: SourceFileLike): Node[] {
         this.assertHasRealPosition("Node without a real position cannot be scanned and thus has no token nodes - use forEachChild and collect the result if that's fine");
-        return this._children || (this._children = createChildren(this, sourceFile));
+        return getOrSetNodeChildren(this, () => createChildren(this, sourceFile));

```

Comment:
I'm not sure I'm a fan of the closure here. Could we modify `setNodeChildren` to return it's input and do this instead?

```suggestion
        return getNodeChildren(this) ?? setNodeChildren(this, createChildren(this, sourceFile));
```

Reply:
I can just do that, yeah. This is the only use.

Reply:
Done.